tools/link.py

```diff
@@ -2052,10 +2052,23 @@ def phase_emit_tsd(options, wasm_target, js_target, js_syms, metadata):
 @ToolchainProfiler.profile_block('embind aot js')
 def phase_embind_aot(options, wasm_target, js_syms):
   out = run_embind_gen(options, wasm_target, js_syms, {})
+
   if DEBUG:
-    write_file(in_temp('embind_aot.js'), out)
+    write_file(in_temp('embind_aot.json'), out)
+  out = json.loads(out)
   src = read_file(final_js)
-  src = do_replace(src, '<<< EMBIND_AOT_OUTPUT >>>', out)
+  src = do_replace(src, '<<< EMBIND_AOT_INVOKERS >>>', out['invokers'])
+  if settings.MODULARIZE == 'instance':
+    # Add ES module exports for the embind exports.
+    decls = '\n'.join([f'export var {name};' for name in out['publicSymbols']])
+    # Assign the runtime exports from Module to the ES export.
+    assigns = '\n'.join([f'{name} = Module[\'{name}\'];' for name in out['publicSymbols']])
+    exports = f'''
+// start embind exports
+() => {{ {assigns} }};
```

Comment:
This looks like an anonymous function that is never called.. maybe I'm missing something?

Reply:
The function gets assigned to the library symbol`embindAssignExports`. It's kind of a hack to include the declarations after it.